manager/allocator/cnmallocator/networkallocator.go

```diff
@@ -142,6 +142,7 @@ func (na *cnmNetworkAllocator) Allocate(n *api.Network) error {
 		n.DriverState = &api.Driver{
 			Name: d.name,
 		}
+		n.IPAM = &api.IPAMOptions{Driver: &api.Driver{}}
```

Comment:
Actually, maybe it's okay if `n.IPAM` is nil before allocation. I guess previous versions would not attempt to use it on an unallocated network. Can you confirm?